### PR TITLE
Fix SVG bugs

### LINKED ISSUE #867: thunk doesn't seem to work with svg child elements
My render functions looks like this:

```javascript

function renderSvg (lines) {
    return h(
        'svg',
        {
            style: {
                height: '200px',
                width: '200px',
                'background-color': 'white'
            },
            attrs: {
                xmlns: 'http://www.w3.org/2000/svg',
                viewBox: '0 0 200 200'
            }
        },
        renderLineGroup(lines)
    )
}


function renderLineGroup (lines) {
    const strokeWidth = 1
    const dashArray = '4 2'

    return h(
        'g',
        {},
        lines.map((l, idx) => thunk('line', `line_${idx}`, renderLine, [ l.x1, l.x2, l.y1, l.y2, strokeWidth, dashArray ]))
        //lines.map((l, idx) => renderLine(l.x1, l.x2, l.y1, l.y2, strokeWidth, dashArray))
    )
}


function renderLine (x1, x2, y1, y2, strokeWidth, dashArray) {
    return h('line', {
        style: {
            'stroke-width': strokeWidth,
            stroke: 'red'
        },
        attrs: { x1, x2, y1, y2, 'stroke-dasharray': dashArray }
    })
}

```

This constructs the svg tag as expected, but it doesn't actually draw. So strange! Here's what it looks like when rendering via `thunk`:
thunk

when I comment out the `lines.map` line and uncomment the one that doesn't use thunk, I see what I'd expect:

non-thunk

### PR DESCRIPTION
**Pull request by macarc**
Fixes #867 and #388

### CODE REVIEW

**Comment by macarc** [comment]
I could also move `addNS` to its own file, since it's not really specific to `h.ts` anymore. Let me know.

File: src/thunk.ts [hunk 1]
```diff
@@ -16,12 +16,14 @@ export interface ThunkFn {
 }
 
 function copyToThunk(vnode: VNode, thunk: VNode): void {
+  const ns = thunk.data ? thunk.data.ns : null;
```
Comment (jvanbruegge) [inline src/thunk.ts:19]:
```suggestion
  const ns = thunk.data?.ns;
```

File: src/tovnode.ts [hunk 2]
```diff
@@ -24,7 +25,16 @@ export function toVNode(node: Node, domApi?: DOMAPI): VNode {
     for (i = 0, n = elmChildren.length; i < n; i++) {
       children.push(toVNode(elmChildren[i], domApi));
     }
-    return vnode(sel, { attrs }, children, undefined, node);
+    const data = { attrs };
+    if (
+      sel[0] === "s" &&
+      sel[1] === "v" &&
+      sel[2] === "g" &&
+      (sel.length === 3 || sel[3] === "." || sel[3] === "#")
```
Comment (jvanbruegge) [inline src/tovnode.ts:33]:
```suggestion
      sel === "svg" ||
      sel.startsWith("svg.") ||
      sel.startsWith("svg#")
```
A bit easier to read

**Review by jvanbruegge — COMMENTED** [review]
Thanks for the PR!
Code looks good, just two minor style comments.

File: src/tovnode.ts (hunk 2 above)
Comment (macarc) [reply src/tovnode.ts:33]: Do you want me to change that here too? https://github.com/snabbdom/snabbdom/blob/9de2307dd9bec6106f587a18eb7b0c994e24722b/src/h.ts#L79-L84

Comment (jvanbruegge) [reply src/tovnode.ts:33]: yes, please

Comment (macarc) [reply src/tovnode.ts:33]: Cool, do you want me to create a new commit or to squash?

Comment (jvanbruegge) [reply src/tovnode.ts:33]: Thinking a bit more about this, this version is a bit more performant. I applied my other suggestion and will merge it directly now

**Comment by jvanbruegge** [comment]
Released as `v3.3.1`

**Pull request merged**